src/emu/emuopts.h

```diff
@@ -212,26 +212,26 @@ class slot_option
 	const std::string &bios() const { return m_specified_bios; }
 	const std::string &default_card_software() const { return m_default_card_software; }
 	bool specified() const { return m_specified; }
-	core_options::entry *option_entry() const { return m_entry; }
+	core_options::entry::shared_ptr option_entry() const { return m_entry; }
 
 	// seters
 	void specify(std::string &&text);
 	void set_bios(std::string &&text);
 	void set_default_card_software(std::string &&s);
 
 	// instantiates an option entry (don't call outside of emuopts.cpp)
-	core_options::entry::ptr setup_option_entry(const char *name);
+	core_options::entry::shared_ptr setup_option_entry(const char *name);
 
 private:
 	void possibly_changed(const std::string &old_value);
 
-	emu_options &           m_host;
-	bool                    m_specified;
-	std::string             m_specified_value;
-	std::string             m_specified_bios;
-	std::string             m_default_card_software;
-	std::string             m_default_value;
-	core_options::entry *   m_entry;
+	emu_options &                   m_host;
+	bool                            m_specified;
+	std::string                     m_specified_value;
+	std::string                     m_specified_bios;
+	std::string                     m_default_card_software;
+	std::string                     m_default_value;
+	core_options::entry::shared_ptr m_entry;
```

Comment:
I thought we agreed this should be a weak pointer since this isn't the owner of the entry.  Making it a shared pointer changes the actual ownership semantics.